gtsfm/averaging/rotation/shonan.py

```diff
@@ -56,23 +69,58 @@ def run(self, num_images: int, i2Ri1_dict: Dict[Tuple[int, int], Optional[Rot3]]
 
         for (i1, i2), i2Ri1 in i2Ri1_dict.items():
             if i2Ri1 is not None:
-                between_factors.append(
-                    BetweenFactorPose3(
-                        i2,
-                        i1,
-                        Pose3(
-                            i2Ri1,
-                            np.zeros(
-                                3,
-                            ),
-                        ),
-                        noise_model,
-                    )
-                )
+                # ignore translation during rotation averaging
+                i2Ti1 = Pose3(i2Ri1, np.zeros(3))
+                between_factors.append(BetweenFactorPose3(i2, i1, i2Ti1, noise_model))
 
         obj = ShonanAveraging3(between_factors, shonan_params)
 
         initial = obj.initializeRandomly()
         result_values, _ = obj.run(initial, self._p_min, self._p_max)
 
-        return [result_values.atRot3(i) for i in range(num_images)]
+        return [result_values.atRot3(i) if result_values.exists(i) else None for i in range(num_connected_nodes)]
+
+    def run(self, num_images: int, i2Ri1_dict: Dict[Tuple[int, int], Optional[Rot3]]) -> List[Optional[Rot3]]:
+        """Wrapper that re-orders keys to prepare a connected graph, w/ N keys ordered [0,...,N-1].
+
+        Args:
+            num_images: number of poses.
```

Comment:
why dont we call it num_poses then? The documentation seems odd.

Reply:
"number of images" would be okay

Reply:
updated, thanks